### DISCUSSION TITLE
Help with substream YAML

### DISCUSSION THREAD

**Discussion by templeowls21** [blakeblackshear/frigate#13948] [Ask A Question]
I'm trying to add a substream to my Frigate YAML in HA to record the HD feed, and use the low res for rtmp.

After adding in the attached code, the add-on won't start. 

When it i delete either just the low or high feed, both work fine. I'm at a loss. 

Any suggestions? Is my syntax wrong?

![Capture](https://github.com/user-attachments/assets/c4c19bf2-9379-415c-b757-da8267dbae4d)

**Comment by NickM-27** [comment]
Rtmp role is no longer supported, go2rtc needs to be used

**Reply by templeowls21** [reply]
Gotcha. Only for substreams? Because rtmp works fine with just one stream path

**Reply by NickM-27** [reply]
because with a single stream path `rtmp` is ignored. `rtmp` as a role does not exist in the latest version of frigate as it says in the release notes

**Reply by templeowls21** [reply]
gotcha...just to be safe, should i still use substreams with go2rtc? Doing some quick research, a few people suggest using one single stream and then use go2rtc to change the resolution to low for live viewing, instead of using a seperate substream. I just want to make sure I'm doing the most efficient method

**Reply by NickM-27** [reply]
if you want to view the sub stream then there is no reason to restream the main stream in go2rtc

**Reply by templeowls21** [reply]
Thanks so much....last question, in my screenshot, I changed 'rtmp' to 'go2rtc' and it worked. But would is it better, efficiently wise, to organize the code like [this](https://docs.frigate.video/configuration/restream#reduce-connections-to-camera)? Where the go2rtc streams are all compiled at the top of the code

**Reply by NickM-27** [reply]
Not sure what you mean by changed rtmp to go2rtc. go2rtc is not a role either.
